test/BoostedMasterChef.test.ts

```diff
@@ -0,0 +1,302 @@
+    this.minter = this.signers[6]
+
+    this.MCV2 = await ethers.getContractFactory("MasterChefJoeV2")
+    this.BMC = await ethers.getContractFactory("BoostedMasterChefJoe")
+
+    this.JoeToken = await ethers.getContractFactory("JoeToken")
+    this.VeJoeToken = await ethers.getContractFactory("VeJoeToken")
+    this.ERC20Mock = await ethers.getContractFactory("ERC20Mock", this.minter)
+    this.SushiToken = await ethers.getContractFactory("SushiToken")
+
+    this.devPercent = 200
+    this.treasuryPercent = 200
+    this.investorPercent = 100
+    this.lpPercent = 1000 - this.devPercent - this.treasuryPercent - this.lpPercent
+    this.joePerSec = 100
+    this.secOffset = 1
+    this.tokenOffset = 1
+  })
+
+  beforeEach(async function () {
+    this.joe = await this.JoeToken.deploy()
+    await this.joe.deployed()
+    this.chef2 = await this.MCV2.deploy(
+      this.joe.address,
+      this.dev.address,
+      this.treasury.address,
+      this.investor.address,
+      this.joePerSec,
+      0,
+      this.devPercent,
+      this.treasuryPercent,
+      this.investorPercent
+    )
+    await this.chef2.deployed()
+
+    await this.joe.transferOwnership(this.chef2.address)
+
+    this.dummyToken = await this.ERC20Mock.connect(this.alice).deploy("Joe Dummy", "DUMMY", 1)
+    this.veJoe = await this.VeJoeToken.connect(this.dev).deploy()
+    await this.chef2.add("100", this.dummyToken.address, ADDRESS_ZERO)
+
+    this.bmc = await this.BMC.deploy(this.chef2.address, this.joe.address, this.veJoe.address, 0)
+    await this.bmc.deployed()
+
+    await this.veJoe.setBoostedMasterChefJoe(this.bmc.address)
+    await this.dummyToken.approve(this.bmc.address, 1)
+    await this.bmc.init(this.dummyToken.address)
+
+    // TODO: Don't name this sushi.
+    this.lp = await this.ERC20Mock.deploy("LPToken", "LP", "10000000000")
+    await this.lp.deployed()
+
+    await this.lp.transfer(this.alice.address, "1000")
+    await this.lp.transfer(this.bob.address, "1000")
+    await this.lp.transfer(this.carol.address, "1000")
+
+    this.bmc.add(100, this.lp.address, ADDRESS_ZERO)
+  })
+
+  it("should adjust boost balance when deposit", async function () {
+    let pool
+    // User has no veJoe
+    await this.lp.connect(this.alice).approve(this.bmc.address, "1000")
+    await this.bmc.connect(this.alice).deposit(0, "1000")
+    pool = await this.bmc.poolInfo(0)
+    expect(pool.totalVeJoe).to.equal(0)
+    expect(pool.totalBoostedAmount).to.equal("1000")
+    expect((await this.bmc.userInfo(0, this.alice.address)).veJoeBalance).to.equal(0)
+
+    // Transfer some vejoe to bob
+    await this.veJoe.connect(this.dev).mint(this.bob.address, 100)
+
+    // Bob enters the pool
+    await this.lp.connect(this.bob).approve(this.bmc.address, "1000")
+    await this.bmc.connect(this.bob).deposit(0, "1000")
+    pool = await this.bmc.poolInfo(0)
+    expect(pool.totalVeJoe).to.equal(100)
+    expect(pool.totalBoostedAmount).to.equal("3500")
```

Comment:
any reason these are strings instead of just numbers? (i.e. `3500` instead of `"3500"`)

Reply:
Definitely not a good reason, I will either change these to be homogeneous or use `ethers.utils.parseEther`